ARCHITECTURE.md

```diff
@@ -0,0 +1,52 @@
+## CHAPTER 1
+
+> **Functional Requirements**
+> 
+> Functional requirements describe the system’s behavior: what it does,
+> the user interactions, and the business logic.
+> 
+> Define the problem, why solving it is relevant, and any vital context.
+> Make a list of questions about the business rules you want answered.
+> 
+> If you have a long list of functional requirements, split them by
+> context (e.g., consumer/logistics or deployment/scaling). Solutions
+> will usually follow these splits, so you may delegate the
+> implementation of context’s features to different teams or just place
+> them in different services.
+
+**1. Função principal - Descoberta de arquivos e diretórios**
+
+-   A aplicação precisa pegar uma URL fornecida e testar possíveis arquivos e diretórios nela.
+-   Se encontrar uma página do tipo “Index Of”, em que já esteja listado os arquivos e diretórios, não é necessário continuar com as tentativas (a maioria das ferramentas não tem essa opção e eu senti falta disso, é útil para laboratórios). Eu gostaria que isso fosse até um padrão, mas com opção de poder desligar caso se encontre falsos positivos.
+-   Ao encontrar um diretório, gostaria de ter a opção da aplicação começar a testar o brute force nele também, para encontrar subdiretórios e arquivos.
```

Comment:
It would be nice to think beyond traditional hosting server path structures (like when Apache and Nginx are used for file hosting). Since I didn't find any suitable article to talk about it, you read an introduction in the following ChatGPT chat: https://chatgpt.com/share/6705d21b-4dbc-8009-9719-3c4e65c582cd